### PR TITLE
Prep 3.1.0 release

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY

Prep 3.1.0 release

Run add_docs, generate changelog

### CODE REVIEW

**Comment by markuman** [comment]
Must the docs be generated against `stable-3` branch?

**Comment by tremble** [comment]
> Must the docs be generated against `stable-3` branch?

Yeah, otherwise the version metadata will be wrong.  What we will need to do is "forward-port" the changelog related updates into main.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -5,6 +5,37 @@ community.aws Release Notes
...
+- aws_s3 - add latest choice on ``overwrite`` parameter to get latest object on S3 (https://github.com/ansible-collections/amazon.aws/pull/595).
+- ec2_vol - add support for OutpostArn param (https://github.com/ansible-collections/amazon.aws/pull/597).
+- ec2_vol - tag volume on creation (https://github.com/ansible-collections/amazon.aws/pull/603).
+- ec2_vpc_route_table - add support for IPv6 in creating route tables (https://github.com/ansible-collections/amazon.aws/pull/601).
+- s3_bucket - Add `validate_bucket_name` option, to control bucket name validation.
+
+Deprecated Features
+-------------------
+
+- ec2_instance - The default value for ```instance_type``` has been deprecated, in the future release you must set an instance_type or a launch_template".
```
Comment (tremble) [inline CHANGELOG.rst:25]: This should definitely link off to the PR.  This must have been missed in the original PR, but I think for deprecation PRs it's more important to include a link so people can understand the background.

**Comment by alinabuzachis** [comment]
> Yeah, otherwise the version metadata will be wrong. What we will need to do is "forward-port" the changelog related updates into main.

@tremble  Not sure how to do that.

**Comment by tremble** [comment]
> > Yeah, otherwise the version metadata will be wrong. What we will need to do is "forward-port" the changelog related updates into main.
> 
> @tremble Not sure how to do that.

Probably a cherry-pick with some things stripped out.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -5,6 +5,41 @@ community.aws Release Notes
...
 
+v3.1.0
+======
+
+Minor Changes
+-------------
+
+- add new parameters hostvars_prefix and hostvars_suffix for inventory plugins aws_ec2 and aws_rds (https://github.com/ansible-collections/amazon.aws/issues/535).
+- aws_s3 - Add ``validate_bucket_name`` option, to control bucket name validation (https://github.com/ansible-collections/amazon.aws/pull/615).
+- aws_s3 - Add `validate_bucket_name` option, to control bucket name validation.
```
[the remark this answers is not in the document]
Comment (alinabuzachis) [reply CHANGELOG.rst]: Thanks. Done that.

**Comment by alinabuzachis** [comment]
> > > Yeah, otherwise the version metadata will be wrong. What we will need to do is "forward-port" the changelog related updates into main.
> > 
> > 
> > @tremble Not sure how to do that.
> 
> Probably a cherry-pick with some things stripped out.

Ah thanks. The thing that brought me to overthink was that "forward-port" 😂

**Review by tremble — APPROVED** [review]

**Review by markuman — APPROVED** [review]

**Pull request merged**